compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -628,6 +628,44 @@ fn check_lhs_nt_follows(sess: &ParseSess, def: &ast::Item, lhs: &mbe::TokenTree)
     // after parsing/expansion. we can report every error in every macro this way.
 }
 
+fn is_empty_token_tree(sess: &ParseSess, seq: &mbe::SequenceRepetition) -> bool {
+    if seq.separator.is_some() {
+        return false;
+    }
+    use mbe::TokenTree;
```

Comment:
Use statements should, with few exceptions (like wildcard enum variant imports), always be on the top of the file. This is not one of the exceptions.

Reply:
The reason for using `use mbe::TokenTree;` here should be to prevent confusion with (rustc_ast::tokenstream::TokenTree)[https://github.com/rust-lang/rust/blob/master/compiler/rustc_ expand/src/mbe/macro_rules.rs#L13], as you can see, there are many uses of `macro_rules.rs`.